os/drivers/lcd/mipi_lcd.c

```diff
@@ -399,6 +408,34 @@ static int lcd_setcontrast(FAR struct lcd_dev_s *dev, unsigned int contrast)
 	return OK;
 }
 
+FAR void lcd_set_logo(FAR struct lcd_dev_s *dev)
+{
+	int logo_arr_index = 0;
+	int lcd_data_index = LCD_LOGO_LOCATION;
+	int lcd_data_col_count = 0;
+	FAR struct mipi_lcd_dev_s *priv = (FAR struct mipi_lcd_dev_s *)dev;
+
+#if !defined(CONFIG_LCD_SW_ROTATION)
+	lcd_data_logo = (uint8_t *)kmm_malloc(CONFIG_LCD_XRES * CONFIG_LCD_YRES * 2 + 1);
+#endif
+	for (int i = 0; i < CONFIG_LCD_XRES * CONFIG_LCD_YRES * 2; i++) {
+		lcd_data_logo[i] = 0;
+	}
+
+	while (logo_arr_index <= (LCD_LOGO_YRES * LCD_LOGO_XRES * 2)) {
+		lcd_data_logo[lcd_data_index] = lcd_logo_data[logo_arr_index++];
+		lcd_data_logo[lcd_data_index + 1] = lcd_logo_data[logo_arr_index++];
+		lcd_data_index += 2;
+		lcd_data_col_count += 1;
+		if (lcd_data_col_count == LCD_LOGO_XRES) {
+			lcd_data_index += ((CONFIG_LCD_XRES - LCD_LOGO_XRES) * 2);
+			lcd_data_col_count = 0;
+		}
+	}
+
+	priv->config->lcd_put_area((u8 *)lcd_data_logo, 1, 1, 480, 800);
```

Comment:
480, 800 should be config_lcd_xres and yres.

Reply:
done